src/v/s3/client.h

```diff
@@ -32,12 +32,42 @@
 #include <chrono>
 #include <initializer_list>
 #include <limits>
+#include <string_view>
 
 namespace s3 {
 
-struct object_tag {
-    ss::sstring key;
-    ss::sstring value;
+/// Object tag formatter that can be used
+/// to format tags for x_amz_tagging field
+/// The value is supposed to be cached and
+/// not re-created on every request.
+class object_tag_formatter {
+public:
+    object_tag_formatter() = default;
+
+    object_tag_formatter(
+      std::initializer_list<std::pair<const char*, const char*>>&& il) {
```

Comment:
Can this be a pair of `std::string_view`s?

Reply:
fixed